corehq/apps/ota/tests/test_search_claim_endpoints.py

```diff
@@ -139,3 +143,115 @@ def test_search_endpoint(self):
+                                "filtered": {
+                                    "filter": {
+                                        "term": {"case_properties.key": "is_inactive"}
+                                    },
+                                    "query": {
+                                        "match": {
+                                            "case_properties.value": {
+                                                "fuzziness": "0", "query": "yes"
+                                            }
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    },
+                    {
+                        "nested": {
+                            "path": "case_properties",
+                            "query": {
+                                "filtered": {
+                                    "filter": {
+                                        "term": {"case_properties.key": "awaiting_claim"}
+                                    },
+                                    "query": {
+                                        "match": {
+                                            "case_properties.value": {
+                                                "fuzziness": "0", "query": "yes"
+                                            }
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                ]
+            }
+        }
+
+        query_partial = {
+            "query": {
+                "filtered": {
+                    "query": {
+                        "bool": {
+                            "must": [
+                                new_must_clause
+                            ]
+                        }
+                    }
+                }
+            }
+        }
+
+        query_addition = CaseSearchQueryAddition(domain=DOMAIN, name="foo", query_addition=query_partial)
+        query_addition.save()
+
+        client = Client()
+        client.login(username=USERNAME, password=PASSWORD)
+        url = reverse('remote_search', kwargs={'domain': DOMAIN})
+        some_case_name = "wut"
+        response = client.get(url, {'name': some_case_name, 'case_type': CASE_TYPE, SEARCH_QUERY_ADDITION_KEY: query_addition.id})
+
+        self.assertEqual(response.status_code, 200)
+
+        expected_query = {
+            'query': {
+                'filtered': {
+                    'filter': {
+                        'and': [
+                            {'term': {'domain.exact': DOMAIN}},
+                            {'term': {'type.exact': CASE_TYPE}},
+                            {'term': {'closed': False}},
+                            {'match_all': {}},
+                            {'match_all': {}}
```

Comment:
Good catch Simon. This was a slight bug resulting from how I was manipulating the `ESQuery` object. `ESQuery.raw_query` gets some clauses added to it based on the state of the ESQuery object. Therefore, these two things aren't equivalent:
```python
raw = es_query_instance.raw_query
es_query_instance.es_query = raw
# Following statement is false
es_query_instance.raw_query == raw
```
I've fixed that erroneous assumption by using `ESQuery.set_query()` instead (and adding a `get_query()` method): https://github.com/dimagi/commcare-hq/pull/14670/commits/dce66d73419e3c875c4ee930caea078da6a96361.